### PR TITLE
WIP: manage a microservice application using traefik load balancer

### PR DESCRIPTION
**Pull request by allamand**
Hi,

I would like to add this Post showing how we can leverge the use of Traefik loadbalancer with docker swarm cluster.

With the recent change of scaling play-with-docker I didn't figure out how I can pre-set urls in the tutorial ideally the user could click on urls that can be changed dynamically like it is done on Katacoda.

If the content seams ok for you I can create more PR with : 
- Manage a microservice using vfarcic docker flow proxy
- Create a cassandra cluster on swarm
- Create a cassandra cluster on swarm and monitore it with Prometheus

Don't hesitate to comment on these.

Sébastien

### CODE REVIEW

**Comment by marcosnils** [comment]
@allamand yes, yes yes!. Thx a lot for this post, been wanting something like this for a while. 

> With the recent change of scaling play-with-docker I didn't figure out how I can pre-set urls in the tutorial ideally the user could click on urls that can be changed dynamically like it is done on Katacoda.

I don't quite understand this, the training site already supports populating links to exposing services like this: ` [This is a link](#){:data-term=".term1"}{:data-port="8888"}`. Is this what you need?


PS: Regarding content streams, please keep them coming!.

**Comment by allamand** [comment]
Hi @marcosnil,

I've missed this feature, seams to be exactly what I was looking for. I'll test that asap, thanks!

Sébastien

**Comment by allamand** [comment]
@marcosnils that was definitely what I was looking for great job ;) I think I would like to extend it a little (addying uri in the end of the url) is that already possible ?

I was including this modification based on latest version on my pwd instance, but I get some errors in sdk.js  `var sessionData = JSON.parse(resp.responseText);`

My PWD return an instance id, while your PWD server return a json.. I try fetch scaling branch but still get an ID instead of a Json, how can I fix this ? (sorry to write thi in the PR log, I can move it if needed)

**Comment by marcosnils** [comment]
> I think I would like to extend it a little (addying uri in the end of the url) is that already possible ?

I don't think so, but it makes totally sense. Maybe something like this could work `[This is a link](/my/uri){:data-term=".term1"}{:data-port="8888"}`.

> My PWD return an instance id, while your PWD server return a json.. I try fetch scaling branch but still get an ID instead of a Json, how can I fix this ? (sorry to write thi in the PR log, I can move it if needed)

Hmmm check your PWD instance again, we're returning a JSON from all the ajax `new_session` calls. https://github.com/franela/play-with-docker/blob/master/handlers/new_session.go#L37-L41

**Comment by allamand** [comment]
@marcosnils the link does not works with URI.
Perhaps we could add a rewrite of the href of the link to the new target

**Comment by marcosnils** [comment]
> the link does not works with URI.

@allamand when I say that **it could work**, I mean that we could implement it that way. It's not currently supported.

**Comment by allamand** [comment]
Ok got it ;)

**Comment by allamand** [comment]
@marcosnils I've added the possibility to make links with `[This is a link](/my/uri){:data-term=".term1"}{:data-port="8888"}`, and updated my 2 blog posts.

Note that link with [This is a link](#){:data-term=".term1"}{:data-port="8888"} won't works because the # is replaced with current location. We must used instead

[This is a link](/){:data-term=".term1"}{:data-port="8888"}

I've corrected earlier blog in the repo that was using that

**Comment by marcosnils** [comment]
@allamand amazing!. I'll test it locally and merge!

File: _posts/2017-04-01-docker-flow-proxy.markdown [hunk 1]
```diff
@@ -0,0 +1,505 @@
+---
+layout: post
+title:  "WebApps with Docker Flow Proxy"
+date:   2017-04-01 10:51:47 +0530
+author: "Sébastien Allamand (allamand)"
+categories: webapps
```
Comment (marcosnils) [inline _posts/2017-04-01-docker-flow-proxy.markdown:6]: `categories` should be `category` and needs to be one of ["beginner", "intermediate", "advanced"]

File: _posts/2017-03-31-traefik-load-balancing.markdown [hunk 2]
```diff
@@ -0,0 +1,437 @@
+---
+layout: post
+title:  "WebApps with Traefik LoadBalancing"
+date:   2017-03-31 10:51:47 +0530
+author: "Sébastien Allamand (allamand)"
+categories: webapps
```
Comment (marcosnils) [inline _posts/2017-03-31-traefik-load-balancing.markdown:6]: `categories` should be `category` and needs to be one of ["beginner", "intermediate", "advanced"]

**Review by marcosnils — COMMENTED** [review]
@allamand PR is amazing. Just added two minor tagging comments.

**Comment by marcosnils** [comment]
@allamand I'll change this myself and merge to master as it's only a small detail.

**Comment by allamand** [comment]
@marcosnils Thanks a lot It was my first merged PR ;)

**Comment by marcosnils** [comment]
@allamand and it was awesome!. Please, keep them coming!.

**Pull request merged**